File: semantic-epa-model/src/main/java/de/fzi/cep/sepa/model/impl/staticproperty/Option.java
package de.fzi.cep.sepa.model.impl.staticproperty;

import javax.persistence.Entity;

import com.clarkparsia.empire.annotation.Namespaces;
import com.clarkparsia.empire.annotation.RdfProperty;
import com.clarkparsia.empire.annotation.RdfsClass;

import de.fzi.cep.sepa.model.UnnamedSEPAElement;

@Namespaces({"sepa", "http://sepa.event-processing.org/sepa#",
	 "dc",   "http://purl.org/dc/terms/"})
@RdfsClass("sepa:Option")
@Entity
public class Option extends UnnamedSEPAElement {

	@RdfProperty("sepa:hasName")
	String name;
	
	@RdfProperty("sepa:isSelected")
	boolean selected;
	
	public Option()
	{
		super();
	}
	
	public Option(String name)
	{
		super();
		this.name = name;
	}
	
	public Option(String name, boolean selected)
	{
		super();
		this.name = name;
		this.selected = selected;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public boolean isSelected() {
		return selected;
	}

	public void setSelected(boolean selected) {
		this.selected = selected;
	}
	
	
}

File: src/app/connect/model/Option.ts
import { RdfId } from '../../platform-services/tsonld/RdfId';
import { RdfProperty } from '../../platform-services/tsonld/RdfsProperty';
import { RdfsClass } from '../../platform-services/tsonld/RdfsClass';

@RdfsClass('sp:Option')
export class Option {
  @RdfId
  public id: string;

  @RdfProperty('sp:elementName')
  public elementName: string;

  @RdfProperty('sp:hasName')
  public name: string;

  @RdfProperty('sp:internalName')
  public internalName: string;

  @RdfProperty('sp:isSelected')
  public selected: boolean;
}
